packages/flutter_test/lib/src/binding.dart

```diff
@@ -974,6 +1011,25 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
     });
   }
 
+  @override
+  Future<void> resolveFutures(Duration expectedIdlePeriod) async {
+    // We must be in the real async zone.
+    assert(_pendingAsyncTasks != null, 'You should not call resolveFutures outside of `runAsync` callback.');
+    const Duration checkInterval = Duration(milliseconds: 50);
+    final Completer<void> resolveFuturesCompleter = Completer<void>();
+    Zone.root.createPeriodicTimer(checkInterval, (Timer timer) {
+      assert(_pendingAsyncTasks != null, 'You should await the result of resolveFutures in `runAsync` callback.');
+      _realAsyncTimerCache.retainWhere((Timer timer) => timer.isActive);
+      if (DateTime.now().difference(_realAsyncLatestMicrotaskCreationTime) > expectedIdlePeriod
```

Comment:
this seems to be encoding a race condition into the test framework, which is a no-no per the style guide